src/server/acl/acl_commands_def.h

```diff
@@ -47,108 +47,21 @@ enum AclCat {
   JSON = 1ULL << 31
 };
 
-// See definitions for NONE and ALL in facade/acl_commands_def.h
-
-inline const absl::flat_hash_map<std::string_view, uint32_t> CATEGORY_INDEX_TABLE{
-    {"KEYSPACE", KEYSPACE},
-    {"READ", READ},
-    {"WRITE", WRITE},
-    {"SET", SET},
-    {"SORTEDSET", SORTEDSET},
-    {"LIST", LIST},
-    {"HASH", HASH},
-    {"STRING", STRING},
-    {"BITMAP", BITMAP},
-    {"HYPERLOG", HYPERLOGLOG},
-    {"GEO", GEO},
-    {"STREAM", STREAM},
-    {"PUBSUB", PUBSUB},
-    {"ADMIN", ADMIN},
-    {"FAST", FAST},
-    {"SLOW", SLOW},
-    {"BLOCKING", BLOCKING},
-    {"DANGEROUS", DANGEROUS},
-    {"CONNECTION", CONNECTION},
-    {"TRANSACTION", TRANSACTION},
-    {"SCRIPTING", SCRIPTING},
-    {"BLOOM", BLOOM},
-    {"FT_SEARCH", FT_SEARCH},
-    {"THROTTLE", THROTTLE},
-    {"JSON", JSON},
-    {"ALL", ALL},
-    {"NONE", NONE}};
-
-// bit 0 at index 0
-// bit 1 at index 1
-// bit n at index n
-inline const std::vector<std::string> REVERSE_CATEGORY_INDEX_TABLE{
-    "KEYSPACE",  "READ",      "WRITE",     "SET",       "SORTEDSET",  "LIST",        "HASH",
-    "STRING",    "BITMAP",    "HYPERLOG",  "GEO",       "STREAM",     "PUBSUB",      "ADMIN",
-    "FAST",      "SLOW",      "BLOCKING",  "DANGEROUS", "CONNECTION", "TRANSACTION", "SCRIPTING",
-    "_RESERVED", "_RESERVED", "_RESERVED", "_RESERVED", "_RESERVED",  "_RESERVED",   "_RESERVED",
-    "BLOOM",     "FT_SEARCH", "THROTTLE",  "JSON"};
-
-// bit index to index in the REVERSE_CATEGORY_INDEX_TABLE
-using CategoryToIdxStore = absl::flat_hash_map<uint32_t, uint32_t>;
-
-inline const CategoryToIdxStore& CategoryToIdx(CategoryToIdxStore store = {}) {
-  static CategoryToIdxStore cat_idx = std::move(store);
-  return cat_idx;
-}
-
-using RevCommandField = std::vector<std::string>;
-using RevCommandsIndexStore = std::vector<RevCommandField>;
-
 constexpr uint64_t ALL_COMMANDS = std::numeric_limits<uint64_t>::max();
 constexpr uint64_t NONE_COMMANDS = std::numeric_limits<uint64_t>::min();
 
-// A variation of meyers singleton
-// This is initialized when the constructor of Service is called.
-// Basically, it calls this functions within the AclFamily::Register
-// functions which has the number of all the acl families registered
 inline size_t NumberOfFamilies(size_t number = 0) {
   static size_t number_of_families = number;
   return number_of_families;
 }
 
-inline const RevCommandsIndexStore& CommandsRevIndexer(RevCommandsIndexStore store = {}) {
-  static RevCommandsIndexStore rev_index_store = std::move(store);
-  return rev_index_store;
-}
+using CategoryIndexTable = absl::flat_hash_map<std::string_view, uint32_t>;
```

Comment:
I see you gave up on using global variables at all. It's ok but that's not what I meant. Sometimes global variables make sense to avoid passing them around especially if their contents stay constant during the run of the program.

I meant that we should not fill data into dynamically allocated global variables before the program starts.

Reply:
in other words you could keep them, but just initialize them during the start sequence of the service.

Reply:
Sure, but `google style of code` is against using global variables with `non trivial destructors`. Here is an example from the docs your referenced on the issue 

```
// bad: non-trivial destructor
const std::string kFoo = "foo";

// Bad for the same reason, even though kBar is a reference (the
// rule also applies to lifetime-extended temporary objects).
const std::string& kBar = StrCat("a", "b", "c");

void bar() {
  // Bad: non-trivial destructor.
  static std::map<int, int> kData = {{1, 0}, {2, 0}, {3, 0}};
}
```

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Do I agree with this rule? I do not (because if you know when you first call `bar()` everything is fine). But since we are following the guidelines I thought it would be best to apply them.

Reply:
p.s.2 if I had to pass them around in many places I would have used the idiom above but it was only a handful one that made me lean toward removing the globals all together

Reply:
ok, sounds good.